server/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -366,13 +365,22 @@ public static boolean validFileName(String fileName) {
     public static boolean validFileNameExcludingAstrix(String fileName) {
         for (int i = 0; i < fileName.length(); i++) {
             char c = fileName.charAt(i);
-            if (c != '*' && INVALID_FILENAME_CHARS.contains(c)) {
+            if (c != '*' && isInvalidFileNameCharacter(c)) {
                 return false;
             }
         }
         return true;
     }
 
+    private static boolean isInvalidFileNameCharacter(char c) {
+        switch (c) {
```

Comment:
I think this method can be written a little more succinctly with a switch expression:

```
return switch(c) {
            case '\\', '/', '*', '?', '"', '<', '>', '|', ' ', ',' -> true;
            default -> false;
        };
```

Reply:
++ done